### PR TITLE
Payments Sync

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/quickbooks_online/migrations/0008_auto_20210120_1831.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                ('updated_at', models.DateTimeField(auto_now=True, help_text='Updated at')),
+            ],
+            options={
+                'db_table': 'bill_payments',
+            },
+        ),
+        migrations.AddField(
+            model_name='bill',
+            name='paid_on_qbo',
+            field=models.BooleanField(default=False, help_text='Payment status in NetSuite'),
```
Comment (ashwin1111) [inline apps/quickbooks_online/migrations/0008_auto_20210120_1831.py:38]: lol

File: apps/quickbooks_online/models.py [hunk 2]
```diff
@@ -145,6 +145,8 @@ class Bill(models.Model):
     currency = models.CharField(max_length=255, help_text='Bill Currency')
     private_note = models.TextField(help_text='Bill Description')
     bill_number = models.CharField(max_length=255)
+    payment_synced = models.BooleanField(help_text='Payment synced status', default=False)
+    paid_on_qbo = models.BooleanField(help_text='Payment status in NetSuite', default=False)
```
Comment (ashwin1111) [inline apps/quickbooks_online/models.py:149]: lol

File: apps/quickbooks_online/models.py [hunk 3]
```diff
@@ -644,3 +646,120 @@ def create_journal_entry_lineitems(expense_group: ExpenseGroup):
...
+
+        vendor_id = None
+        if expense_group.fund_source == 'PERSONAL':
+            vendor_id = Mapping.objects.get(
+                source_type='EMPLOYEE',
+                destination_type='VENDOR',
+                source__value=description.get('employee_email'),
+                workspace_id=expense_group.workspace_id
+            ).destination.destination_id
+        elif expense_group.fund_source == 'CCC':
```
Comment (Shwetabhk) [inline apps/quickbooks_online/models.py:701]: Why are we creating Bill Payments for CCC?

File: apps/quickbooks_online/models.py [hunk 4]
```diff
@@ -644,3 +646,120 @@ def create_journal_entry_lineitems(expense_group: ExpenseGroup):
...
+            vendor_id = general_mappings.default_ccc_vendor_id
+
+        general_mappings = GeneralMapping.objects.get(workspace_id=expense_group.workspace_id)
+        bill_payment_object, _ = BillPayment.objects.update_or_create(
+            expense_group=expense_group,
+            defaults={
+                'private_note': 'Reimbursable expenses by {0}'.format(description.get('employee_email')) if
+                expense_group.fund_source == 'PERSONAL' else
+                'Credit card expenses by {0}'.format(description.get('employee_email')),
```
Comment (Shwetabhk) [inline apps/quickbooks_online/models.py:710]: Why Credit Card Expenses?

File: apps/quickbooks_online/tasks.py [hunk 5]
```diff
@@ -603,3 +604,251 @@ def create_journal_entry(expense_group, task_log):
...
+
+
+def create_bill_payment(workspace_id):
+    fyle_credentials = FyleCredential.objects.get(workspace_id=workspace_id)
+
+    fyle_connector = FyleConnector(fyle_credentials.refresh_token, workspace_id)
+
+    fyle_connector.sync_reimbursements()
+
+    bills = Bill.objects.filter(
```
Comment (Shwetabhk) [inline apps/quickbooks_online/tasks.py:628]: If we're filtering by Fund source personal here why add an else clause in model bill payment function?

File: apps/quickbooks_online/models.py (hunk 3 above)
Comment (Sravanksk) [reply apps/quickbooks_online/models.py:701]: This is incorrect have updated this

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**